src/rust/queues/src/lib.rs

```diff
@@ -0,0 +1,6 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+pub mod mpsc;
```

Comment:
There are a couple differences between the `mpsc` and `spsc` traits beside `waker`. I wonder what is the reason for this asymmetry?

Reply:
I think migrating to a bidirectional queue for all the queues in the server probably makes sense - we can simplify the queues crate to have just that type. I'll clean that up

Reply:
This crate is refactored and I have consolidated on a single paired queue type